components/automate-cli/cmd/chef-automate/helper.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+	"github.com/chef/automate/components/automate-cli/pkg/docs"
+	"github.com/chef/automate/components/automate-cli/pkg/status"
+	"github.com/chef/automate/components/automate-deployment/pkg/cli"
+	"github.com/spf13/cobra"
+)
+
+var nodeDeleteCmd = &cobra.Command{
+	Use:               "node-delete [uuid]",
+	Short:             "Delete node by node uuid",
+	Long:              "",
+	PersistentPreRunE: checkLicenseStatusForExpiry,
+	RunE:              runDeleteNodeCmd,
+	Args:              cobra.ExactArgs(1),
+	Annotations: map[string]string{
+		docs.Tag: docs.BastionHost,
+	},
+}
+
+type LicenseResult struct {
+	Result           LicenseStatus `json:"result"`
+	ErrorType        string        `json:"error_type"`
+	ErrorDescription string        `json:"error_description"`
+}
+
+type LicenseStatus struct {
+	CustomerName   string         `json:"customer_name"`
+	LicenseType    string         `json:"license_type"`
+	LicenseId      string         `json:"license_id"`
+	ExpirationDate ExpirationDate `json:"expiration_date"`
+	GracePeriod    bool           `json:"grace_period"`
+}
+
+type ExpirationDate struct {
+	Seconds int64 `json:"seconds"`
+}
+
+func checkLicenseStatusForExpiry(cmd *cobra.Command, args []string) error {
+	fileName := "/tmp/license"
+	err := commandPrePersistent(cmd)
+	if err != nil {
+		return status.Wrap(err, status.CommandExecutionError, "unable to set command parent settings")
+	}
+
+	cmd1 := exec.Command("chef-automate", "license", "status", "--result-json", fileName)
+	// Not checking for error, as if the license is expired it will still return error and for commercial license a grace period is required
+	cmd1.Output()
+
+	licenseResult, err := readFileAndMarshal(fileName)
+	if err != nil {
+		return err
+	}
+
+	err = checkLicenseExpiry(licenseResult)
```

Comment:
We need to have two functions here, one will return error and one will return warning.

Only for infrastructure we need warning. Calling both the functions will always give you error first and exit the command.

For all other commands, warning is required and command should run as it is.